src/extensions/CustomDateUtils.tsx

```diff
@@ -0,0 +1,75 @@
+import { DateIOFormats } from '@date-io/core/IUtils';
+import DayjsUtils from '@date-io/dayjs';
+import dayjs, { Dayjs } from 'dayjs';
+
+interface Opts {
+    locale?: string;
+    instance?: typeof dayjs;
+    formats?: Partial<DateIOFormats>;
+  }
+
+  type Constructor<TDate extends Dayjs> = (
+    ...args: Parameters<typeof dayjs>
+  ) => TDate;
+
+const withLocale = <TDate extends Dayjs>(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    dayjs: any,
```

Comment:
This looks like it's going to be `Dayjs`, or `typeof dayjs`

Reply:
Tried this, didn't work too well -- I can show you tomorrow